### ISSUE TITLE
Unable to click anything until action key is pressed again

### ISSUE THREAD

**Issue by Agamennon** [mlutfy/hidetopbar#51]
sometimes after regular use i cant click anything in my desktop, not until i resort to manualy toggoling overview with the windows key. (i use 4 monitors)  but in gnome 3.12 it was not a issue

**Comment by GNOMES** [comment]
iI am having the same issue on Arch running Gnome 3.14. I currently have user themes, insensitive message tray and caffeine. I see the other reported issue with Chrome, usually occurs when web browsing with a few tabs open. I use three screens if that helps. Running Autohide with default settings.

**Comment by GNOMES** [comment]
Any logs that might help?

**Comment by tuxor1337** [comment]
I don't think there are any helpful logs. The only helpful thing at the moment would be to have clear step by step instructions on how to reproduce this.

**Comment by GNOMES** [comment]
It happens on my laptop at home when connected to my extra monitors, and as well at school with just my laptop screen. 

I have my bar autohide all the time except in overview. I have chrome open with 3-4 tabs, and always when switching tabs. Cant think of it occurring with other windows. I am currently using google-chrome-beta from the AUR

No really way to duplicate that I can tell. Its hard to tell, but it almost seems that the active window is almost underneath where the top bar would be, even tough the top bar is closed.. Just like the other user reported above my post, cant click on anything until super key is hit. I ran TOP after this occures and gnome shell memory usage sky rockets when this happens. Sometimes when it occurs Chromes colors are dimmed as if I clicked on a different window, and chrome is no longer the active window.


I mispoke in my last post, I have the following settings setup:

Show panel when mouse near edge - off
keep hot corner active in hidden state - on
in the above case - off
pressure barrier threshold - 200
pressure arrier timeout - 200
animation - .1
aimation - .1
key that triggers the bar - disabled
delay - .1
pressing hte shortcut - on
only hide the panel - off

I made these settings to eliminate having impatient animation gnome shell extension

**Comment by tuxor1337** [comment]
Hm. I'm so sorry, but I don't have time to install Chrome and use it for hours just to trigger (and analyze) this issue :-/

**Comment by GNOMES** [comment]
Understandable. Any possibly ideas what it could be? Could it be an issue with Chrome looking for the top of the window, and going past where it should be?

**Comment by tuxor1337** [comment]
Hm, I have no idea. Since I started maintaining this extension, I was more or less solving problems by trial and error. I don't really have a good understanding of how all this works ;-)

**Comment by GNOMES** [comment]
Okay so after working on my laptop all day today, I can for sure say I only had this problem after using dual screens. Still with chrome. After working in another screen long enough, chrome(open on screen with top bar of course) become inactive. It only seems to occur when I click on the tabs instinctively to switch tabs.

**Comment by dffischer** [comment]
I also experience this problem. It normally occurs when working with the cursor near the top, such as web browsing with tabs. Unfortunately, I am also unable to provide any directions on how to reproduce it. But I can tell that
- it is not related to the number of screens, as I only use a single monitor and never attached a second display,
- it is not related to Chrome, as I experience it on Firefox regularly and think I remember a single time it occurred while reordering terminals, and
- a recent change seems to improve things. I am usually lazy and update my extensions only when a new shell version is released. Since the update to shell version 3.16 not long ago, the bug appears significantly less often, but is not gone entirely.

As a stab in the dark, I will set my `animation-time-autohide` to the ridiculously high value of two seconds, observe whether this changes anything about the next few days and report back.

**Comment by tuxor1337** [comment]
@dffischer GNOME 3.16 is not supported by hide top bar as of now...

**Comment by dffischer** [comment]
I am aware that 3.16 support has not has not landed yet. But reading about your plans to implement it in the future and as this and #49 also occur on previous versions it couldn't hurt to gather some data. Apart from these two issues, the extension serves me very well on 3.16.

**Comment by tuxor1337** [comment]
Now, that there is 3.16 support and that there have been significant changes to the extension's behaviour with todays update, are there any news on this issue? I still can't reproduce it, even though I'm working hours and hours with a maximized Firefox window and a lot of tab switching.

If there's no way to reproduce this, I will decide at some point to close this, because there's really no point to it...

**Comment by GNOMES** [comment]
The issues are on Chrome

**Comment by tuxor1337** [comment]
@dffischer reported that the problems are equivalently present with Firefox.

Can you, GNOMES, reproduce this issue? Is it still present for you with the latest version of this extension? Are you on GNOME 3.14 or 3.16?

**Comment by GNOMES** [comment]
Sorry that email was not supposed to send yet! It was a draft. I remember
you stating that you so not have time to test chrome. The issue occurred
when using google chrome on a multi monitor setup. The mouse would become
unresponsive. I was on 3.14 when this occured. I just reinstalled Arch with
3.16 on my machine. When I get home I can install the extension and test.
Just wanted to add this as you are using Firefox, might be an issue with
chrome.

**Comment by tuxor1337** [comment]
Via extensions.gnome.org and probably related to this issue in some way:

> Hide Top Bar displaces mouse position for Google Chrome. See https://code.google.com/p/chromium/issues/detail?id=494206 for full details and settings

**Comment by latot** [comment]
how to reproduce, gentoo-64 with chrome and gnome-3.16 affects nautilus too, well open that apps with the extension enabled, then use alt+f2 and run "r" to reload gnome.

**Comment by tuxor1337** [comment]
@latot You seem to describe something else that is also true if all GNOME extensions (including hide top bar) are disabled: If you open a single nautilus window and then restart gnome-shell with Alt+F2 and "r", there is a text box in the center of your screen saying "Restarting...". If, at this point, you click somewhere on the screen _before_ that box fades out, you are trapped in a situation where your nautilus window does not react to mouse clicks etc. Hence: What you describe, is not related to this extension!

Okay, my friends, I'm on GNOME 3.18 and I still haven't been able to reproduce the behavior described in this report. If nobody comes up with clear instructions, I'm closing this.

**Comment by GNOMES** [comment]
If I use only my laptop screen I can switch between windows without issues using Overview. If I have my one VGA monitor and my HDMI monitor connected and switch windows on different monitors chrome becomes unresponsive. Only way to fix is to alt f2 restart gnome like others are saying. 

Still happening using the git version of your app on Arch Linux using Gnome 3.18.3-1

When I got to Settings > Displays > Arrange displays it shows a black bar between my monitors ( I use a one adjacent and one vertical). I am wondering if this issue is due to the fact it appears you are not truly hiding the top bar but rather moving it above the displayed area on the screen ( answering why there is a thick bar of black between my monitors (see screenshot). I have issues with different sized monitors on Gnome as it is as some programs will load above the area of my left monitor. As far as  I can tell gnome still thinks of this area as usable screen space. Not sure if it is trying to use this area between the top and bottom monitors or well? 

Guessing that if you have only one screen or are using uniform monitors that might answer why you are unable to see this issue. 

You are finally using Chrome correct? You were refusing to use it before as your a firefox user. I will try disabling my little monitor on the right and have only side by side monitors to test. I will update with results as both my laptop and monitor I had as a top monitor are both 1920 x 1080

**Comment by tuxor1337** [comment]
You are telling me that this is only happening when using three monitors of different size while Google Chrome is open? That's unfortunate... I have two monitors of different size, I can install Google Chrome, but where am I supposed to get a third monitor? Are you sure, this can't be reduced to a simpler setup?

**Comment by tuxor1337** [comment]
By the way: While trying to reproduce this issue, I found that the switch-workspace animation of GNOME is working very similar to the slide-out animation of Hide Top Bar. Having a monitor on top of the current primary monitor doesn't seem to be considered an option by GNOME developers...

**Comment by GNOMES** [comment]
I can't garentee its due to monitors, just thinking of reasons you haven't
been able to replicate issue

**Comment by latot** [comment]
i only can say, i can reproduce it in one monitor...

**Comment by tuxor1337** [comment]
@latot How? Note that I commented above on your alleged steps to reproduce.

**Comment by latot** [comment]
in my case i write the steps in a comment (up)
```
how to reproduce, gentoo-64 with chrome and gnome-3.16 affects nautilus too, well open that apps with the extension enabled, then use alt+f2 and run "r" to reload gnome.
```
but thinking it feels say 3.16 don't is enough, gnome have a lot of packages and i don't know which cause this.
And as you say this probably don't is a hidetopbar bug, and more specify, i found you can click, but in other position, for example it move, if you click the (1,1) the action click can be in (4,3) are (3,2) of difference, so if you will click the point (5,4) will be in the (8,6),  knowing that i normaly can continuos working.
But when i test it the condition to can reproduce this is have this extension enabled, and chrome with 3 or 4 tabs.

**Comment by GNOMES** [comment]
I have been using my machine with one monitor for troubleshooting. This screen is from when it happened. You can see that by clicking on the tabs of chrome it seems to click "through" chrome as to where the gnome bar would be. 

How are you implementing this extension? Are you just moving the top bar above the screen, or are you minimizing it? Would adding a string preventing all clicks or say removing it from top bar competently except in overview fix issue?
![screenshot from 2015-12-03 23-16-35](https://cloud.githubusercontent.com/assets/7462046/11582401/0cf91b76-9a14-11e5-89de-5f628c85dea2.png)

**Comment by tuxor1337** [comment]
As mentioned above: "Here are the details on how Hide Top Bar currently works: It moves the panel out of the primary monitor's area. After the animation is over, it makes the panel transparent and reduces its height to 1."

I see that you are using a GTK theme. Do you also use a gnome-shell theme?

**Comment by dffischer** [comment]
I finally found a way to reproduce it, even with a single monitor! I run Archlinux with gnome shell 3.18 and get to the bug like this.

- Open Firefox.
- Maxmize it.
- Open a new tab so that you now have multiple tabs opened at the top of the window.
- Open any other window, not maxmized, and focus it.
- Target the upper edge of any Firefox tab, where it touches the hidden top bar.
- Click immediately when touching the edge.

It seems to me that the click has to happen when Firefox already recognized the hover event and prepares to show the tooltip for the tab, but the pressure barrier of the top bar has not yet been triggered to move the bar down below the cursor to steal focus from Firefox.

This results in an interface completely unreactive to any click, the tab tooltip stuck above the now visible top bar, until the action key is pressed.

![the bug in action](https://cloud.githubusercontent.com/assets/6497034/11589402/7f30790e-9a88-11e5-810f-35d73318f0d8.png)

I did not test it with Chrome, but it also works with Thunderbird and even gnome-terminal. Seemingly, any application featuring tabs is sufficient to reproduce the bug.

**Comment by tuxor1337** [comment]
This is in need of further clarification:

* Where is Firefox's title bar? Do you use the GNOME extension Maximus? Are there any other GNOME extensions that you use in parallel?
* You enabled mouse-sensitivity of the top bar. What other settings did you set for Hide Top Bar?

**Comment by dffischer** [comment]
Yes, I use [MaximusTwo](https://github.com/wilfm/GnomeExtensionMaximusTwo). I deactivated it for the test, but it seems to stay active nonetheless until the shell is restarted. The bug indeed vanishes when title bars are present, as there are no tabs at the top of windows to click.

So this is a compatibility issue and unrelated to the problem originating from multiple monitors. I opened a separate issue #88 for this problem and also notified [MaximusTwo](https://github.com/wilfm/GnomeExtensionMaximusTwo) and [Pixel Saver](https://github.com/wilfm/GnomeExtensionMaximusTwo), two popular forks of the original Maximus extension.

**Comment by tuxor1337** [comment]
Since Google Chrome doesn't have window decoration either, I suspect that this has to do with the missing title bar.

@dffischer I tried to reproduce what you describe and was able to reproduce it once in 20 or 30 tries. That's probably enough to confirm this bug...

**Comment by tuxor1337** [comment]
Oh, by the way, maybe you should consider using BottomPanel together with Hide Top Bar. That brings the panel to the bottom of the screen so that it doesn't conflict with the firefox tabs on top.

**Comment by dffischer** [comment]
> was able to reproduce it once in 20 or 30 tries
That reflects how often it appears in my daily use. Especially its seldom occurence left me quite stumped the first few times it happened.

I intensively thought about a combination with BottomPanel. Unfortunately, it has [its own set](https://github.com/meghprkh/wintitle/issues/4) of [compatibility issues](https://github.com/Thoma5/gnome-shell-extension-bottompanel/issues/3).

**Comment by tuxor1337** [comment]
Why would you hide the panel and still use the feature that you can unmaximize windows by click-and-dragging on the panel? In this case, I can only recommend to use more keyboard shortcuts. Alt+F10 for unmaximizing, Alt+F7 for moving etc.

**Comment by dffischer** [comment]
I find it rather unintuitive to use mouse and keyboard together. It never appeared as efficient to me to show the panel with a shortcut and then grab the mouse to click something on it. Likewise, I use the mouse to position windows on the screen. Reaching to the keyboard to start this process, especially when the window in question is not currently focused, is quite cumbersome to me.

But this may just be the top notch of maximization to an already quite efficient workflow relevant to few but myself. The crux is, when working with the mouse in gnome-shell, I would like to utilize the bottom edge more.

**Comment by tuxor1337** [comment]
Is this still an issue? I can't test myself, because MaximusTwo doesn't seem to work on GNOME 3.22.

**Comment by dffischer** [comment]
MaximusTwo and some other extensions do work on GNOME 3.22. They are incompatible with Wayland, though. So I am currently running GNOME 3.22 with Xorg, still.

As I am indeed now also using bottompanel, I do not evoke the bug in my daily use. Following the instructions I posted above, I was unable to reproduce it while bottompanel was deactivated.

Perhaps the problem resolved itself, perhaps it will reappear when I am finally able to move to Wayland together with all my favourite extensions. If so, I will report here again.

**Comment by tuxor1337** [comment]
Closing this as "automagically resolved" ;-)
